src/layer/vulkan/shader/unaryop.comp

```diff
@@ -3,40 +3,27 @@
 
 #version 450
 
-layout (constant_id = 0) const int op_type = 0;
+layout (constant_id = 0) const uint n = 0;
+layout (constant_id = 1) const int op_type = 0;
 
-#define shape_constant_id_offset 1
-layout (constant_id = shape_constant_id_offset + 0) const int dims = 0;
-layout (constant_id = shape_constant_id_offset + 1) const int w = 0;
-layout (constant_id = shape_constant_id_offset + 2) const int h = 0;
-layout (constant_id = shape_constant_id_offset + 3) const int c = 0;
-layout (constant_id = shape_constant_id_offset + 4) const int cstep = 0;
-
-layout (binding = 0) buffer bottom_top_blob { sfp bottom_top_blob_data[]; };
+layout (binding = 0) buffer bottom_top_blob { sfpvec4 bottom_top_blob_data[]; };
 
 layout (push_constant) uniform parameter
 {
-    int dims;
-    int w;
-    int h;
-    int c;
-    int cstep;
+    uint n;
+    int op_type;
```

Comment:
`op_type` is always known at compile-time, it is no necessary to be a push constant